presentation/backend/src/presentation-backend/RulesetEmbedder.ts

```diff
@@ -33,6 +33,24 @@ export enum DuplicateRulesetHandlingStrategy {
   Replace,
 }
 
+/** Interface for callbacks which will be called before and after Element updates */
+interface OnElementUpdate {
+  onBeforeElementUpdate: () => Promise<void>;
+  onAfterElementUpdate: () => Promise<void>;
+}
+
+/** Interface for callbacks which will be called before and after Element is inserted */
+interface OnElementInsert {
+  onBeforeElementInsert: () => Promise<void>;
+  onAfterElementInsert: () => Promise<void>;
+}
+
+/** Interface for callbacks which will be called before and after Model is inserted */
+interface OnModelInsert {
+  onBeforeModelInsert: () => Promise<void>;
+  onAfterModelInsert: () => Promise<void>;
+}

```

Comment:
- Missing release tags (use `@beta`)
- I'd rather we used 2 interfaces: `UpdateCallbacks<TProps>` and `InsertCallbacks<TProps>`. `TProps` should be passed to "before" and "after" callback functions.

Reply:
Updated